pkg/apis/openstack/validation/infrastructure.go

```diff
@@ -76,45 +77,63 @@ func ValidateInfrastructureConfigUpdate(oldConfig, newConfig *api.Infrastructure
 func ValidateInfrastructureConfigAgainstCloudProfile(infra *api.InfrastructureConfig, shootRegion string, cloudProfileConfig *api.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if ok, validFloatingPoolNames := validateFloatingPoolNameConstraints(cloudProfileConfig.Constraints.FloatingPools, shootRegion, infra.FloatingPoolName); !ok {
-		allErrs = append(allErrs, field.NotSupported(fldPath.Child("floatingPoolName"), infra.FloatingPoolName, validFloatingPoolNames))
-	}
+	allErrs = append(allErrs, validateFloatingPoolNameConstraints(cloudProfileConfig.Constraints.FloatingPools, shootRegion, infra.FloatingPoolName, fldPath)...)
 
 	return allErrs
 }
 
-func validateFloatingPoolNameConstraints(names []api.FloatingPool, region string, name string) (bool, []string) {
+func validateFloatingPoolNameConstraints(fps []api.FloatingPool, region string, name string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if _, err := findFloatingPool(fps, region, name, fldPath.Child("floatingPoolName")); err != nil {
+		allErrs = append(allErrs, err)
+	}
```

Comment:
```suggestion
	_, err := findFloatingPool(fps, region, name, fldPath.Child("floatingPoolName"))
  allErrs = append(allErrs, err)
```

Reply:
I think it's not desirable to let `allErrs` contain a `nil` element. I changed the `findFloatingPool` to return an `ErrorList` instead.